Refreshing Fred Hutch WDL Content

tefirman opened the pull request:
## Description
- Added new "WDL Workflows" article covering WDL fundamentals
- Converted Cromwell page to "WDL Execution Engines" with expanded coverage of Cromwell/Sprocket/miniWDL and a comparison of which one is best for which use-case
- Overhauled the "Using Workflows" article with WDL vs Nextflow comparison
- Added WDL background section to "PROOF How-To" with links to WILDS WDL Library
- Updated cross-references in SciComp articles (Compute Environments, Parallel Computing, Batch Computing pathway, Software Overview)
- Reordered left sidebar menu for better navigation

## Related Issue
- Further addresses #1111
- Partially addresses #1110 (Docker content refresh soon to come!!
- Fixes #1244 

## Testing
- Built site locally, looks good, tested most links.

emjbishop commented on _datademos/proof-how-to.md:
```diff
@@ -16,67 +16,26 @@ This guide is intended to describe how you can run PROOF, catering to varying le
-> Quick note: the Cromwell server is referred to as a PROOF server in these instructions. PROOF handles setting up the Cromwell server for you.
+> Quick note: Throughout this guide, "PROOF server" and "Cromwell server" refer to the same thing - PROOF just handles all the Cromwell setup for you.
 
 ## Using PROOF
```
Under this section it says to use the Marconi network, is that still true? It should be Research Staff now, right? Line 51

emjbishop commented on _datascience/using_workflows.md:
```diff
@@ -1,39 +1,235 @@
+**Learn more:**
+- [Nextflow at Fred Hutch](/compdemos/nextflow/) - Getting started guide
+- [Nextflow Catalog](/datascience/nextflow_catalog) - Fred Hutch curated workflows
+- [nf-core](https://nf-co.re/pipelines) - Community workflow catalog
```
```suggestion
### WDL (Workflow Description Language)

**Best for:**
- Users who want a simple, human-readable workflow language
- Integration with Fred Hutch infrastructure via PROOF

**Key features:**
- Open-source language developed by the Broad Institute
- [WILDS WDL Library](/datascience/wilds_wdl/) provides tested, reusable WDLs built by and for Fred Hutch scientists
- [PROOF](/datascience/proof/) platform for easy execution on Fred Hutch cluster

**Learn more:**
- [WDL Workflows Guide](/datascience/wdl_workflows/) - Language fundamentals
- [WDL Execution Engines](/datascience/wdl_execution_engines/) - How to run WDL workflows
- [WILDS WDL Library](/datascience/wilds_wdl/) - Ready-to-use modules and workflows
- [PROOF](/datascience/proof/) - User-friendly workflow submission at Fred Hutch

### Nextflow

**Best for:**
- Users comfortable with Groovy/Java-like syntax
- Integration with [nf-core](https://nf-co.re/pipelines) community workflows
- Projects needing fine-grained control over execution

**Key features:**
- Mature ecosystem with extensive community support
- Native support for containers, conda, and modules
- Large collection of pre-built workflows via [nf-core](https://nf-co.re/pipelines)
- Active Fred Hutch community

**Learn more:**
- [Nextflow at Fred Hutch](/compdemos/nextflow/) - Getting started guide
- [Nextflow Catalog](/datascience/nextflow_catalog) - Fred Hutch curated workflows
- [nf-core](https://nf-co.re/pipelines) - Community workflow catalog
```

I would take out lines that are implied or don't help distinguish the two from each other. I would also try to note that with WDL we can take requests for WDL Library additions

emjbishop commented on _datascience/using_workflows.md:
```diff
@@ -1,39 +1,235 @@
+| **Execution options** | Multiple engines (Sprocket, miniWDL, Cromwell) | Nextflow runtime |
+| **Local testing** | Easy with Sprocket/miniWDL | Easy with Nextflow |
+| **Pre-built workflows** | WILDS WDL Library, GATK workflows | nf-core (500+ workflows) |
+| **Language style** | Declarative (what to do) | Imperative (how to do it) |
```
```suggestion
```
I don't think this line adds much to someones decision making

emjbishop commented on _datascience/using_workflows.md:
```diff
@@ -1,39 +1,235 @@
+- **Custom containers** - Build your own Docker images with [WILDS Docker Library](https://github.com/getwilds/wilds-docker-library)
+- **Workflow optimization** - Fine-tune resource requests and parallelization
+- **CI/CD** - Automate workflow testing with GitHub Actions
+- **Publishing** - Share workflows via GitHub for reproducible research
```
```suggestion
```
I would cut this out since there is no guidance offered on these topics and people will come across these organically.

emjbishop commented on _datascience/using_workflows.md:
```diff
@@ -1,39 +1,235 @@
 
-## Getting Started with Nextflow
+**Q: Do I need to learn Docker?**
+A: Basic Docker knowledge is helpful but not required. Both WILDS WDL Library and nf-core provide pre-built containers. Workflow engines handle container pulling automatically.
```
```suggestion
A: Basic Docker knowledge is very helpful. Pre-built Docker containers exist for both WDL and Nextflow and the workflow engines handle container pulling automatically. So while you may not need to build your own container, you are likely to be working with containers.
```
This has been everyone's biggest sticking point when getting started with WDL (including DaSL folks!) so I would just say yes to mentally prepare them

emjbishop commented on _datascience/wdl_execution_engines.md:
```diff
@@ -0,0 +1,272 @@
+
+> **New to WDL?** Start with our [WDL Workflow guide](/datascience/wdl_workflows/) to learn about the language itself before diving into execution engines.
+
+## Execution Engine Comparison
```
Is the purpose of this page to help people choose one? Or is it to provide some education to clear confusion? Having this table up front feels like the audience is IT or something (since regular folks aren't going to use cromwell on their workstations).

I would take out this table or move it further down after some more education.

tefirman replied:
The purpose is more to provide education around WDL execution engines, but to your point, this is probably too much too quick and would be more valuable towards the end as a sort of summary. Moved it down next to the "Decision Guide" at the end in the latest commit.

emjbishop commented on _datascience/wdl_execution_engines.md:
```diff
@@ -0,0 +1,272 @@
+4. Upload your WDL file and inputs JSON
+5. Submit and monitor your workflow through the dashboard
+
+PROOF handles all Cromwell server setup, configuration, and job scheduling automatically.
```
You may want to briefly explain somewhere what it means to 'start a server', since 'server' conjures images of a physical computer rack. I would personally benefit and someone who doesn't know what cromwell is may benefit too

emjbishop commented on _datascience/wdl_execution_engines.md:
```diff
@@ -0,0 +1,272 @@
+**Considerations:**
+- Less feature-rich than Cromwell
+- Primarily focused on local execution
+- Limited backend options compared to Cromwell
```
```suggestion
- Install with `pip install miniwdl`
- Designed for running on your own machine
```
Again, I think just these are the points that add value

emjbishop commented on _datascience/wdl_execution_engines.md:
```diff
@@ -0,0 +1,272 @@
+miniwdl run workflow.wdl input_file=/path/to/file.fastq sample_name=sample1
+
+# Validate WDL without running
+miniwdl check workflow.wdl
```
```suggestion
```

I think they can go to the docs for details

tefirman replied:
Totally fair, consolidated into super basic example in the latest commit under "Getting Started with miniWDL" and pointed to their docs for the rest.

emjbishop commented on _datascience/wdl_execution_engines.md:
```diff
@@ -0,0 +1,272 @@
+
+**For production pipelines:**
+- Use **Cromwell** for call caching, server mode, and robust HPC integration
+- Leverage PROOF if you want infrastructure managed for you
```
```suggestion
## Summary

**For Fred Hutch researchers running on Gizmo or AWS Batch:**
- Use **PROOF** (Cromwell backend). Must use WDL version 1.0.
- Consider testing locally before submitting to PROOF/Cromwell

**For local workflow development and testing:**
- Use **Sprocket** for actively maintained modern execution
- Use **miniWDL** as an alternative if Sprocket doesn't meet your needs
```

I think this can be simplified

tefirman replied:
Agreed, way too repetitive now that I re-read it. Made these changes in the latest commit.

emjbishop commented on _datascience/wdl_execution_engines.md:
```diff
@@ -0,0 +1,272 @@
+- Use WDL version 1.0 (required for PROOF)
+- Specify Docker containers (not institution-specific modules)
+- Use standard WDL features, avoiding engine-specific extensions
+- Test on multiple engines if sharing workflows broadly
```
```suggestion
**All WDLs in the [WILDS WDL Library](/datascience/wilds_wdl/) have been tested on Cromwell, miniWDL, and Sprocket.**
```

I think this is the only important point here

emjbishop reviewed (commented):
Overall this is an amazing amount of great work and is a wonderful overhaul of the sciwiki info related to WDL. At times it feels almost **too** comprehensive to the point of being redundant or creating confusion.

I do think `wdl_workflows.md` needs some major rethinking ("WDL workflows" is a big topic) - is this meant to be a sales pitch, a tutorial, a reference? I would consider the purpose of this page and its target audience and modify it to suit just that one purpose with those people in mind. Los of linking out may be necessary.

tefirman commented on _datascience/wdl_execution_engines.md:
```diff
@@ -0,0 +1,272 @@
+2. Navigate to [proof.fredhutch.org](https://proof.fredhutch.org)
+3. Log in with your Hutch credentials
+4. Upload your WDL file and inputs JSON
+5. Submit and monitor your workflow through the dashboard
```
Great call, actually think that should go before WDL upload in step 4. Added in the latest commit!

tefirman commented on _datascience/wdl_execution_engines.md:
```diff
@@ -0,0 +1,272 @@
+miniwdl --version
+
+# Check WDL syntax
+miniwdl check workflow.wdl
```
Totally fair, consolidated into super basic example in the latest commit under "Getting Started with miniWDL" and pointed to their docs for the rest.

tefirman commented on _datascience/wdl_execution_engines.md:
```diff
@@ -0,0 +1,272 @@
+Sprocket automatically:
+- Pulls required Docker containers
+- Manages input/output files
+- Creates run directories with results and logs
```
Yeah, this is probably too much too soon and they can go to the WILDS WDL Library docs for more details on how to use those if they want. Switched to a more simple example as a "Getting Started with Sprocket" section in the latest commit.

tefirman commented on _datascience/wdl_workflows.md:
```diff
@@ -0,0 +1,367 @@
+
+The Workflow Description Language (WDL) is an open-source language for describing data processing workflows with human-readable syntax. WDL makes it straightforward to define analysis tasks, chain them together in workflows, and parallelize their execution across different computing environments.
+
+## Why Use WDL?
```
I think it's at least partially a sales pitch in that people want to know more about WDL before pulling the trigger between that or Nextflow, so I think the "Why Use WDL" section is important to explain what the language has to offer before diving into details. Mind if leave this in?

tefirman commented:
@emjbishop -- Addressed all of your review comments above, really appreciate your feedback!! I did push back a bit on the updates to `wdl_workflows.md`, but huge improvements across the board. Mind taking one more quick look and approving if all looks good?

sitapriyamoorthi commented on _datascience/cirro.md:
```diff
@@ -96,7 +96,7 @@ batch1_sampleD,batch1_sampleD_R1.fastq.gz,batch1_sampleD_R2.fastq.gz,control
 
 In addition to a catalog of [pre-configured workflows](https://docs.cirro.bio/pipelines/overview/#pipelines-in-cirro),
-Cirro can be used to run custom [WDL](/compdemos/Cromwell/) or [Nextflow](https://www.nextflow.io/) workflows.
+Cirro can be used to run custom [WDL workflows](/datascience/wdl_workflows/) or [Nextflow](https://www.nextflow.io/) workflows.
```
Do we want to add a link to Cirro website?

tefirman replied:
Absolutely, added links for the Cirro SciWiki article to the "Using Workflows" and "WDL Workflows" articles. Great call!

sitapriyamoorthi commented on _datascience/using_workflows.md:
```diff
@@ -1,39 +1,184 @@
+| **Community resources** | Growing (WILDS, Broad, Terra) | Extensive (nf-core, global community) |
+| **Execution options** | Multiple engines (Sprocket, miniWDL, Cromwell) | Nextflow runtime |
+| **Local testing** | Easy with Sprocket/miniWDL | Easy with Nextflow |
+| **Pre-built workflows** | WILDS WDL Library, [GATK workflows](https://github.com/gatk-workflows) | nf-core (500+ workflows) |
```
BioWDL?

sitapriyamoorthi commented on _datascience/using_workflows.md:
```diff
@@ -1,39 +1,184 @@
+| Consideration | WDL | Nextflow |
+|--------------|-----|----------|
+| **Learning curve** | Gentle - simple, declarative syntax | Moderate - Groovy/Java-like syntax |
+| **Fred Hutch integration** | Excellent via PROOF | Excellent via direct execution |
```
Cirro?

sitapriyamoorthi commented on _datascience/wdl_execution_engines.md:
Looks great! Very helpful article!

sitapriyamoorthi reviewed (approved):
Looks great! Huge PR! Awesome content to help people understand WDLs. Left a few comments for you to address. Approving it now.

tefirman commented on _compdemos/aws-batch.md:
```diff
@@ -10,7 +10,7 @@ SciComp provides direct access to AWS Batch in two ways:
 * Via programmatic interfaces such as Python's [boto3](https://boto3.readthedocs.io/en/latest/reference/services/batch.html#client). 
 
->Note:  For the great majority of users, using AWS Batch is best done by using a workflow manager such as [Cromwell](/compdemos/Cromwell/) or[Nextflow](/compdemos/nextflow/) both of which are supported at the Hutch and can allow you to leverage cloud computing resources such as AWS Batch without creating a workflow that will ONLY work on AWS Batch.  
+>Note: For the great majority of users, using AWS Batch is best done by using a workflow manager such as [WDL workflows](/datascience/wdl_workflows/) (executed via [Cromwell or other engines](/datascience/wdl_execution_engines/)) or [Nextflow](/compdemos/nextflow/), both of which are supported at the Hutch and can allow you to leverage cloud computing resources such as AWS Batch without creating  ...
```
Oooooo, love that idea, added it as an issue: https://github.com/FredHutch/wiki/issues/1302

tefirman commented on _datascience/using_workflows.md:
```diff
@@ -1,39 +1,184 @@
+
+**Choose WDL if you:**
+- Are new to workflow systems
+- Want to use PROOF for easy cluster submission
```
Absolutely, added in https://github.com/FredHutch/wiki/pull/1276/commits/40b00e5cec187e5b8bd794258782daec2ab6048c

tefirman commented on _datascience/using_workflows.md:
```diff
@@ -1,39 +1,184 @@
-## Getting Started with WDL
+1. **Start small** - Test with small datasets before scaling up
+2. **Use version control** - Track your workflows in Git
+3. **Containerize** - Use Docker/Apptainer for reproducibility
```
Good catch, added links for both the Docker and Apptainer articles in https://github.com/FredHutch/wiki/pull/1276/commits/40b00e5cec187e5b8bd794258782daec2ab6048c

tefirman commented on _scicomputing/compute_jobs.md:
```diff
@@ -15,14 +15,14 @@ This section is intended to be a basic introduction to using the workload manage
 - monitoring jobs and providing you with metadata about them and the workflow itself. 
 
-Two workflow managers in use on the Fred Hutch campus are [Nextflow](/compdemos/nextflow/) and [Cromwell](/compdemos/Cromwell/) and users are actively curating more shared support and resources at those pages as well as in GitHub.  Workflow manager related information is collected as a GitHub [Workflow Manager Project](https://github.com/orgs/FredHutch/projects/8) as well as specific [Nextflow re ...
+At Fred Hutch, [Nextflow](/compdemos/nextflow/) and [WDL workflows](/datascience/wdl_workflows/) are the primary workflow systems in use. WDL workflows can be executed using [multiple engines](/datascience/wdl_execution_engines/) including Cromwell, miniWDL, and Sprocket, while Nextflow has its own execution engine. Users are actively curating shared support and resources, with the [WILDS WDL Lib ...
```
Totally fair, switched the language to "main workflow systems recommended" in https://github.com/FredHutch/wiki/pull/1276/commits/1b295f558af5cabbb79f81249c1cef3ac09112da
